Bump docker-compose postgres image to 13

maeve-fpf opened the pull request:
Per web sync, let's upgrade this to match our infrastructure. This one is a little different due to the slightly customized image that Mike created a while back. All it does is add a script to enable `hstore`, which seems like something that app migrations can handle. Maybe we can remove it?

maeve-fpf committed:
Bump docker-compose postgres image to 13

chigby commented:
I'm pretty sure it can be done with migrations.. want me to add that to this PR?

chigby committed:
Create hstore extension via migrations

Previously we were doing this via a script that worked slightly
differently on CI and for local development, and this simplifies that
entire situation.

chigby committed:
Use latest available version of docker on CI

Fixes a problem where the root user did not have permissions in the
postgres 13 container.  See
https://stackoverflow.com/q/69706677/1382355

chigby reviewed (approved):
Looks good, and now CI is passing due to the addition of a `version` argument to the `setup_remote_docker` command.
